### PR TITLE
build: add Go 1.23 to unit tests CI

### PR DESCRIPTION
**Pull request by haoxins**
#### What type of PR is this?
build

#### Check the PR title.

- [x] This PR title match the format: \<type\>(optional scope): \<description\>
- [x] The description of this PR title is user-oriented and clear enough for others to understand.
- [ ] Attach the PR updating the user documentation if the current PR requires user awareness at the usage level. [User docs repo](https://github.com/cloudwego/cloudwego.github.io)

#### (Optional) More detailed description for this PR(en: English/zh: Chinese).

en:
zh(optional):

### CODE REVIEW

**Comment by haoxins** [comment]
```
link: github.com/bytedance/mockey/internal/monkey/common: invalid reference to runtime.sysAllocOS
```

`github.com/bytedance/mockey` needs support Go 1.23 :)

**Comment by haoxins** [comment]
> ```
> link: github.com/bytedance/mockey/internal/monkey/common: invalid reference to runtime.sysAllocOS
> ```
> 
> `github.com/bytedance/mockey` needs support Go 1.23 :)

CI should be passed now.

File: .github/workflows/tests.yml [hunk 1]
```diff
@@ -34,15 +34,15 @@ jobs:
           cache: false # don't use cache for self-hosted runners
 
       - name: Unit Test
-        run: go test -race -covermode=atomic -coverprofile=coverage.txt ./...
+        run: go test -race -ldflags=-checklinkname=0 -covermode=atomic -coverprofile=coverage.txt ./...
```
Comment (GuangmingLuo) [inline .github/workflows/tests.yml:37]: flag provided but not defined: -checklinkname

**Comment by haoxins** [comment]
the CI failed seems due to GitHub being down~

File: .github/workflows/tests.yml (hunk 1 above)
Comment (haoxins) [reply .github/workflows/tests.yml:37]: yeah, emmm, this is 1.23 only

Comment (haoxins) [reply .github/workflows/tests.yml:37]: still better to wait for `github.com/bytedance/mockey`'s new version.

**Comment by haoxins** [comment]
Using `mockey v1.2.11`
https://github.com/bytedance/mockey/pull/61

**Comment by GuangmingLuo** [comment]
在 Windows 环境下 ，mockey 还不支持 go 1.23

**Comment by GuangmingLuo** [comment]
Please update `mockey v1.2.12`

**Comment by haoxins** [comment]
> Please update `mockey v1.2.12`

done

**Review by GuangmingLuo — APPROVED** [review]

**Review by welkeyever — APPROVED** [review]

**Pull request merged**